### PR TITLE
Improve non-windows dev experience and toolchain update

### PR DESCRIPTION
**Pull request by tonyredondo**
This PR improves the non-windows developer experience by: 

- Simplifying the steps to build the native profiler. Just cmake . && make. All dependencies are downloaded on demand and works with linux and macos.
- Also updates the toolchain to a recent version of llvm, clang. We bump the ubuntu version to the latest LTS 20.04 release.
- Removes multiple copies of spdlog includes and uses a single copy for all architectures. 

@DataDog/apm-dotnet

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Native/class_factory.cpp [hunk 1]
```diff
@@ -56,11 +56,21 @@ HRESULT STDMETHODCALLTYPE ClassFactory::CreateInstance(IUnknown* pUnkOuter,
...
+
+#ifdef AMD64
+            , " (amd64)"
+#elif X86
+            , " (x86)"
+#elif ARM64
+            , " (arm64)"
+#elif ARM
+            , " (arm)"
+#endif
```
Comment (lucaspimentel) [inline src/Datadog.Trace.ClrProfiler.Native/class_factory.cpp:73]: The "64-bit" or "32-bit" a few lines above is now redundant. Should we remove those?

**Review by lucaspimentel — COMMENTED** [review]
Happy to see this finally getting some love.

File: src/Datadog.Trace.ClrProfiler.Native/class_factory.cpp (hunk 1 above)
Comment (tonyredondo) [reply src/Datadog.Trace.ClrProfiler.Native/class_factory.cpp:73]: If you prefer, I can remove it yes. My main goal is to add the architecture in the logs, but you're right, is redundant.

Comment (tonyredondo) [reply src/Datadog.Trace.ClrProfiler.Native/class_factory.cpp:67]: done in https://github.com/DataDog/dd-trace-dotnet/pull/1096/commits/1a080be596797fe841d1f10873e7b129b2472534

**Review by lucaspimentel — APPROVED** [review]
Thanks!

**Comment by zacharycmontoya** [comment]
@tonyredondo can you remove the `.DS_Store` files and update our repo's .gitignore to ignore them?

**Comment by tonyredondo** [comment]
> @tonyredondo can you remove the `.DS_Store` files and update the repo's `.gitignore` to ignore them?

Sure, sorry for that I forgot the macos hidden files...

**Comment by tonyredondo** [comment]
> @tonyredondo can you remove the `.DS_Store` files and update the repo's `.gitignore` to ignore them?

Done in 1fb7c58

**Comment by tonyredondo** [comment]
@zacharycmontoya

**Review by zacharycmontoya — APPROVED** [review]
This is awesome, thanks for the improvements!

**Pull request merged**